### PR TITLE
Support for short RGB

### PR DESCRIPTION
**Pull request by JohnDowson**

### CODE REVIEW

File: aos_head_generator/main.py [hunk 1]
```diff
@@ -16,10 +16,13 @@ def create_head(color):
...
 @app.route("/v1/heads/<string>.png")
 def send_image(string):
-    if len(string) != 6:
+    if len(string) not in (6, 3):
         abort(400)
 
-    color_str = string[0:2], string[2:4], string[4:6]
+    l = len(str)//3
+    color_str = string[0:l], string[l:l*2], string[l*2:l*3]
+    color_str = map(lambda x: if len(x) < 2: x+x, color_str)
```
Comment (NotAFile) [inline aos_head_generator/main.py]: This is invalid syntax. This line also seems unnecessary? `int(x, 16)` works with both 1 and 2 length strings. List comprehensions are also preferable to `map(lambda x: ...)` as they are a lot easier to read.

I'd suggest testing your code locally before submitting a PR.

**Review by NotAFile — CHANGES_REQUESTED** [review]

File: aos_head_generator/main.py (hunk 1 above)
Comment (JohnDowson) [reply aos_head_generator/main.py]: It is necessary as otherwise it would only set lower bytes for each color.

File: aos_head_generator/main.py [hunk 2]
```diff
@@ -16,10 +16,13 @@ def create_head(color):
...
 @app.route("/v1/heads/<string>.png")
 def send_image(string):
-    if len(string) != 6:
+    if len(string) not in (6, 3):
         abort(400)
 
-    color_str = string[0:2], string[2:4], string[4:6]
+    l = len(str)//3
+    color_str = string[0:l], string[l:l*2], string[l*2:l*3]
+    color_str = (x+x if len(x) < 2 else x for x in color_str)
```
Comment (NotAFile) [inline aos_head_generator/main.py]:
```suggestion
    # double up single-length values
    color_str = (x+x if len(x) < 2 else x for x in color_str)
```

**Comment by NotAFile** [comment]
That last commit inproved things a bunch, thanks.

**Pull request merged**